addons/mail/models/reset_template_mixin.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import os
+
+from lxml import etree
+
+from odoo import api, fields, models, tools
+from odoo.modules import get_module_resource
+from odoo.modules.module import get_resource_from_path, get_resource_path
+from odoo.tools.convert import xml_import
+from odoo.tools.misc import file_open
+from odoo.tools.translate import TranslationFileReader
+
+
+class ResetTemplateMixin(models.Model):
+    _name = "reset.template.mixin"
+    _description = 'Reset Template Mixin'
+
+    template_fs = fields.Char(
+        string='Template Filename', copy=False,
+        help="""File from where the template originates. Useful to reset broken template.""")
+    is_template_modified = fields.Boolean(string='Is Template Modified', copy=False)
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for val in vals_list:
+            if 'install_filename' in self._context:
+                # we store the relative path to the resource instead of the absolute path, if found
+                # (it will be missing e.g. when importing data-only modules using base_import_module)
+                path_info = get_resource_from_path(self._context['install_filename'])
+                if path_info:
+                    val['template_fs'] = '/'.join(path_info[0:2])
+        return super().create(vals_list)
+
+    def write(self, vals):
+        vals['is_template_modified'] = vals.get('is_template_modified', True)
+        return super().write(vals)
+
+    # ------------------------------------------------------------
+    # RESET TEMPLATE
+    # ------------------------------------------------------------
+
+    def _process_translation_data(self, trans_file, lang, xml_ids):
+        """Populates the ir_translation table.
+        :param trans_file: path to open a translation file
+        :param lang: language code of the translations contained in `trans_file`
+                     language must be present and activated in the database
+        """
+        with file_open(trans_file, mode='rb') as fileobj:
+            fileformat = os.path.splitext(trans_file)[-1][1:].lower()
+            reader = TranslationFileReader(fileobj, fileformat=fileformat)
+            # Process a single PO (or POT) entry
+            for row in reader:
+                if row.get('imd_name') in xml_ids:
+                    # copy Translation from Source to Destination object
+                    self.env['ir.translation']._set_ids(
+                        row['name'], 'model', lang, self._ids, row.get('value', ''), row['src'],
```

Comment:
are you sure it's always `model` ? can't we have a field with html_translate (and then `model_terms`) ?